fiftyone/utils/yolo.py

```diff
@@ -1082,24 +1083,30 @@ def _get_yolo_v5_labels_path(image_path):
 
 def _parse_yolo_row(row, classes):
     row_vals = row.split()
-    target, xc, yc, w, h = row_vals[:5]
+    target = row_vals[0]
+    confidence = None
+    if len(row_vals) < 7: # box case
+        xc, yc, w, h = map(float, row_vals[1:5])
+        minx = xc - 0.5 * w
+        miny = yc - 0.5 * h
+
+        if len(row_vals) > 5:
+            confidence = float(row_vals[5])
+
+    else: # polygon case
```

Comment:
@sergiev since this case is polygons, we could just return `fo.Polylines()` instances here instead of `fo.Detections()` and then there would be no data loss at all

Reply:
Thank you sincerely, didn't know that! I'll rewrite this part to polylines and post update here soon after the test.

Reply:
@brimoor sorry, but looks like Polylines and Detections differences too big to fit the scope of this PR.
When I return Polylines instead of Detections the non-interchangeability comes clear.
![image](https://github.com/voxel51/fiftyone/assets/27900857/523c4678-2d7c-45dd-a54a-ec0562429af1)
I'll be happy to fix it, but need guidance: which level (module, file, method) will be the best place to propose such changes?